stacks-signer/src/runloop.rs

```diff
@@ -937,21 +1069,98 @@ impl<C: Coordinator> SignerRunLoop<Vec<OperationResult>, RunLoopCommand> for Run
     }
 }
 
+/// When this threshold is reached, the signer will use the NACKed CoordinatorMetadata if that data
+/// is more updated than its own chain view
+fn get_nack_threshold(total_signers: u32) -> u32 {
+    (total_signers + 1) / 2
+}
+
+/// Clean up outdated coordinator_metadata NACKs that have a stale or equivalent chain view compared to the current signer
+fn cleanup_outdated_coordinator_nacks(
+    nack_messages_received: &mut HashMap<CoordinatorMetadata, HashSet<u32>>,
+    current_observed_height: u64,
+) {
+    let outdated_metadata_to_remove: Vec<CoordinatorMetadata> = nack_messages_received
+        .iter()
+        .filter(|(coordinator_metadata, _)| {
+            coordinator_metadata.stacks_block_height <= current_observed_height
```

Comment:
I'm not exactly sure how to think about this in Nakamoto (chime in anyone) but I think we should wait at least a few blocks until we remove old nacks.

Also, I just learned about [`HashMap::retain`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.retain) which is an easy "iterate and keep some items" method. Tiny impact in this case but it'll simplify the code too.